src/communication/request_sync_client_server.hpp

```diff
@@ -193,31 +194,50 @@ namespace cubcomm
 
     sequenced_payload ip (NO_RESPONSE_SEQUENCE_NUMBER, std::move (a_payload));
 
-    m_queue->push (a_outgoing_message_id, std::move (ip));
+    // NOTE: if needed, errors can be handled
+    m_queue->push (a_outgoing_message_id, std::move (ip), nullptr);
   }
 
   template <typename T_OUTGOING_MSG_ID, typename T_INCOMING_MSG_ID, typename T_PAYLOAD>
-  void
+  css_error_code
   request_sync_client_server<T_OUTGOING_MSG_ID, T_INCOMING_MSG_ID, T_PAYLOAD>::send_recv (
 	  T_OUTGOING_MSG_ID a_outgoing_message_id, T_PAYLOAD &&a_request_payload, T_PAYLOAD &a_response_payload)
   {
     // Get a unique sequence number of response and group with the payload
-    response_sequence_number rsn = m_rsn_generator.get_unique_number ();
+    const response_sequence_number rsn = m_rsn_generator.get_unique_number ();
     sequenced_payload seq_payload (rsn, std::move (a_request_payload));
 
+    send_queue_error_handler error_handler_ftor = [&rsn, this] (
+		css_error_code error_code, bool &abort_further_processing)
+    {
+      abort_further_processing = false;
+      this->m_response_broker.register_error (rsn, std::move (error_code));
+    };
+
     // Send the request
-    m_queue->push (a_outgoing_message_id, std::move (seq_payload));
+    m_queue->push (a_outgoing_message_id, std::move (seq_payload), nullptr /*std::move (error_handler_ftor)*/);
```

Comment:
Why  error_handler ftor is disabled here?

Reply:
Corrected.